package/network/services/dropbear/files/dropbear.init

```diff
@@ -80,7 +80,20 @@ hk_config()
 # $1 - host key file name
 hk_config__keyfile() { hk_config keyfile "$1" ; }
 
-ktype_all='ed25519 ecdsa rsa'
+hk_enabled_ktypes() {
+	local ktypes ktype values value ret
+
+	ktypes=$($PROG --help 2>&1|awk '/\s+- /{print $2}')
+	values=$(uci get dropbear.@dropbear[-1].HostKeyTypes 2>/dev/null)
```

Comment:
What happens when this configuration option is not set? It should then behave like before and generate keys of all supported types.

Reply:
Agreed. Perhaps a fallback or default value. 

Also, run `uci -q` instead to avoid `2>/dev/null`.